File: cpp/Easy/645_set_mismatch.cpp
class Solution {
public:
    vector<int> findErrorNums(vector<int>& nums) {
        vector<int> formArray;
        sort(nums.begin(), nums.end());
        map<int,int> mapa;
        map<int,int> amout;

        for( int i = 0; i < nums.size(); i++){
            amout[nums[i]]++;
            if(amout[nums[i]] == 2){
                formArray.push_back(nums[i]);
            }
        }
        for( int i = 0; i < nums.size(); i++){
            mapa[nums[i]] = i+1;
            if(--amout[mapa[nums[i]]] < 0 ){
                formArray.push_back(mapa[nums[i]]);
            }
        }

        return formArray;
    }
};

File: python/easy/645_set_mismatch.py
class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        l = len(nums)
        mapa = [0]* (l+1)
        miss = 0
        dub = 0
        for n in nums:
            mapa[n]+=1
        for i in range(1, len(mapa)):
            if mapa[i] == 0:
                miss = i
            if mapa[i] == 2:
                dub = i
        return [dub, miss]
